tests/integration/test_record_writer.py

```diff
@@ -299,3 +303,120 @@ def test_decoding_line_numbers(tmp_path):
             ("bar", "bar b.py", 16),
             ("foo", "foo a.py", 5),
         ]
+
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 13), reason="Object tracking requires Python 3.13+"
+)
+def test_write_object_tracking_records(tmp_path):
+    """Test writing object tracking records to a file."""
+    # GIVEN
+    output_file = tmp_path / "test_object_tracking.memray"
+
+    # WHEN
+    writer = RecordWriterTestHarness(
+        str(output_file),
+        native_traces=True,
+        trace_python_allocators=False,
+        track_object_lifetimes=True,
+        file_format=FileFormat.ALL_ALLOCATIONS,
+        main_tid=1,
+    )
+
+    # Write some basic setup records
+    assert writer.write_thread_record(1, "MainThread")
+    assert writer.write_code_object(1, "test_func", "test.py", b"", 1)
+    assert writer.write_frame_push(1, 1, 10, True)
+
+    # Write object tracking records
+    assert writer.write_object_record(1, 0x12345678, True, 0)  # Object created
+    assert writer.write_object_record(
+        1, 0x87654321, True, 1
+    )  # Object created with native frame
+    assert writer.write_object_record(1, 0x12345678, False, 0)  # Object destroyed
+    assert writer.write_object_record(
+        1, 0x87654321, False, 1
+    )  # Object destroyed with native frame
+
+    assert writer.write_trailer()
+
+    # THEN
+    header_fields, records = parse_capture_file(output_file)
+
+    # Check that track_object_lifetimes is in the header
+    header_dict = dict(header_fields)
+    assert header_dict.get("track_object_lifetimes") == "true"
+
+    # Find the object records in the parsed output
+    object_records = [
+        record for record in records if record.startswith("OBJECT_RECORD")
+    ]
+
+    # We should have 4 object records
+    assert len(object_records) == 4
+
+    # Check the object records are correctly formatted
+    assert any("address=0x12345678" in record for record in object_records)
+    assert any(
+        "0x87654320" in record or "0x87654321" in record for record in object_records
+    )
```

Comment:
Please make these make an assertion about the exact text of the record as printed by `memray parse`, like `test_write_basic_records` does it. As written, this is too generic to prove that the test record writer is working correctly. This test would pass if it always wrote created records and never wrote deleted records, for instance.